prog/mutation.go

```diff
@@ -102,9 +102,16 @@ func (p *Prog) Mutate(rs rand.Source, ncalls int, ct *ChoiceTable) {
 						filename := r.filename(s)
 						arg.Data = []byte(filename)
 					case sys.ArrayType:
-						count := r.rand(6)
-						if count == uintptr(len(arg.Inner)) {
-							count++
+						count := uintptr(0)
+						switch a.Kind {
+						case sys.ArrayRandLen:
+							count = r.rand(6)
+							a.Len = count
+						case sys.ArrayFixedLen:
```

Comment:
You also need to update mutationArgs, that function still looks at a.Len to understand if we need to mutate the array or not. I think it's better if we don't add fixed-size arrays to mutation candidates, then you need a panic here.